### ISSUE TITLE
Incorrect figure rendered?

### ISSUE THREAD

**Issue by sriki18** [caiofcm/filter_pandoc_run_py#7]
I am trying to have multiple figures in the same document. For this I use the following modified snippet from one of the test documents:

---START CODE---

```python
#filter: {.run 
# 				caption="Figure Number One" 
# 				label="my_fig" 
# 				}
import matplotlib
matplotlib.use('AGG')
from matplotlib import pyplot as plt
plt.plot([1, 2], [3, 4], 'dr-')
```

### A runnable code with two figures generation

It will run the code and save the created pyplot figure to a file, than it creates a Image text block to load it.

Syntax: \{.python .run format=[blockquote (default), text]\ hide_code=[False, True] caption="" label="" width=""}}

- Uses _pandoc-fignos_ filter notation for caption, label and width.
- If more than one figure is created the configuration keyvals can be added, such as: caption2="" label2="" width2="" ext2="" ...
- Figure is saved to folder `./plt-images`

```python
#filter: {.run caption2="Number One" caption3="Other Figure" label2="my_fig_2" label3="my_fig_3"}
import matplotlib
matplotlib.use('AGG')
from matplotlib import pyplot as plt
from mpl_toolkits.mplot3d import Axes3D
from matplotlib import cm
import numpy as np

# First Figure
plt.plot([1, 2], [3, 4], 'dg-')

# Second Figure
fig = plt.figure()
ax = fig.gca(projection='3d')
X = np.arange(-5, 5, 0.25)
Y = np.arange(-5, 5, 0.25)
X, Y = np.meshgrid(X, Y)
Z = np.sin(np.sqrt(X**2 + Y**2))
surf = ax.plot_surface(X, Y, Z, cmap=cm.coolwarm)
fig.colorbar(surf, shrink=0.5, aspect=5)
```
---END CODE---

## Modifications

In the second plot, I change the color from red to green. I also change some of the filter tags (`caption2`, `caption3`).

## Build command
`pandoc tut.md -F filter_pandoc_run_py -F pandoc-citeproc -o tut.pdf`

## Problem

The second plot should be green and the first plot should be red. Instead, both the second and first plot end up being green. See [tut.pdf](https://github.com/caiofcm/filter_pandoc_run_py/files/2700480/tut.pdf).

**Comment by kiwi0fruit** [comment]
Does this work without `filter_pandoc_run_py`? If yes you can try using [Pandoctools/Knitty/Matplotlib Helper](https://github.com/kiwi0fruit/pandoctools) combo as it doesn't alter Matplotlib behaviour.

**Comment by caiofcm** [comment]
The filter runs each block of code in a single python interpreter. Thus, you don't need the code:

 ```{python}
import matplotlib
matplotlib.use('AGG')
```
in both code blocks (just in the first one suffice).

I was able to reproduce your error. Thank you for letting me know!

I merged a fix for it and also include the option to use the **python
figure title as the caption**.

See file: `./examples/runnable_three_figures_dif_cells.md`, which was
converted with:

`pandoc ./examples/runnable_three_figures_dif_cells.md -F filter_pandoc_run_py  -o out.pdf`

Bests

fix #7
